src/openrct2/world/Location.hpp

```diff
@@ -139,6 +139,11 @@ struct CoordsXY
 
         return rotatedCoords;
     }
+
+    CoordsXY ToTileStart() const
+    {
+        return { x & ~0x1F, y & ~0x1F };
```

Comment:
Would `floor2( x, 32)` be better its slightly clearer why it is doing this.

Reply:
I have mixed feelings, after I got around what `floor2` does, it seems ok to use it instead, but I'm not sure. What's your take @Gymnasiast ?

Reply:
I wonder if `floor2(x, 32)` get properly optimised to `x & ~0x1F` by the compiler. If it doesn't, I prefer `x & 0x1F`.

Another option is adding a comment: `Tiles are divided into 32 x coordinates and 32 y coordinates. We truncate it to a multiple of 32 to go back to the start of the tile.`

Reply:
Floor2 is just a macro doing exactly that. I suspect it's the same

Reply:
I updated it to use `floor2` instead, will submit another PR to improve `floor2` documentation and implementation